### ISSUE TITLE
seleniumhq version 4 support

### ISSUE THREAD

**Issue by rvanderboom** [fhoeben/hsac-fitnesse-fixtures#340]
hey, 

i don't know where to ask a question in stead of create an issue.
My question is when is hsac-fitnesse-fixtures support the 4.0.0-alpha version org.seleniumhq.selenium ?
When i try with latest alpha-6 version the driver setup gives an Unknown Method capabilities 

The reason why i try is i find running headless chrome very unstable with this version.
When running individual tests everything ok, but with large suite, i often get :

org.openqa.selenium.WebDriverException: unknown error: session deleted because of page crash from tab crashed 

I use the latest chrome 83 version.

**Comment by fhoeben** [comment]
I have not tried selenium 4 alpha. 

Do you mean the headless chrome is unstable with selenium 3? Theoretically the chromedriver and chrome combination should determine the exact interaction with chrome, the selenium version should not matter.

I believe the latest version of chrome is 84, and the 4.17.0 includes the driver for that version.

The tab crashes I've seen in the past have mostly been because of (memory) issues in JavaScript code of the website...

If you would have a page/test combination that always produces the error I can investigate further.

**Comment by rvanderboom** [comment]
ok, thanks for the quick response.
I wil try to reproduce the issue, since it happens when running more tests in a Suite together...

Is there already a plan to support selenium alpha 4 in the near future ?

**Comment by fhoeben** [comment]
No immediate plans (but if you have time to experiment I would love to hear your findings).

**Comment by rvanderboom** [comment]
Hey, 

yes i will experience some with the 4 alpha and mabe to make it work overwrite some classes..
This because from beginnen our test suite never was 100% stable...

We have 2 issues sometime:
* invalid session id ...    (see message below).
* test that stops (hangs ?) without any message half way the suite... Also no error log, nothing.
   This while we specify timeouts on every storyboard browsertest. Running second time most of time then 
   goes correctly 
----
 
org.openqa.selenium.NoSuchSessionException: invalid session id
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'fitnesse-0', ip: '10.42.21.40', os.name: 'Linux', os.arch: 'amd64', os.version: '4.14.186-146.268.amzn2.x86_64', java.version: '16-ea'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 83.0.4103.116, chrome: {chromedriverVersion: 83.0.4103.116 (8f0c18b4dca9..., userDataDir: /tmp/.org.chromium.Chromium...}, goog:chromeOptions: {debuggerAddress: localhost:34707}, javascriptEnabled: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: LINUX, platformName: LINUX, proxy: Proxy(), setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: ignore, webauthn:virtualAuthenticators: true}
Session ID: 67b39e58619507eb2551c9af61f34732
*** Element info: {Using=xpath, value=//div[@class='input-message error-message show-input-error']}
at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:64)
at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)

**Comment by rvanderboom** [comment]
mm tried to adjust some hsac class to work with selenium alpha-6, but its to much so many methods change.

That feels like going to be long time before this version is supported ?

**Comment by rvrijhoef** [comment]
Is see this issue is about alpha version, but is still open. Selenium 4.0.0 is now officially released. Is there already a plan to update hsac-fitnesse-fixtures with the latest Selenium?

**Comment by fhoeben** [comment]
There is a PR currently on the stable release #355 
It looks quite promising. I hope @tcnh gets around to polishing the code so we can get it merged.

@rvrijhoef any particular Selenium 4 features you want to use, or just to keep up with progress?

**Comment by rvrijhoef** [comment]
@fhoeben there are some options with the (Chrome)devTools I would like the use. But the PR looks quite promising indeed. Hopefully you can keep up the progress

**Comment by rvanderboom** [comment]
hey, i build latest selenium4 version and tested it with our test suite (headless, 40 browser tests).
The run fine with this version !

**Comment by dineshkumar-kanniyammal** [comment]
org.openqa.selenium.NoSuchSessionException: Tried to run command without establishing a connection
 
 1.Selenium 4.0.0 jar
 2.Firefox 95 verision brower
 3.gecko driver v0.30.0

this problem occured while running parallel Firefox browser
.Can anyone help?

**Comment by fhoeben** [comment]
@dineshkumar-kanniyammal which version of the project are you using? Since 5.0.0 the fixtures use Selenium 4.1.2

If you have a specific problem please create a separate issue for that describing your exact setup (how are you running in parallel, how many sessions etc) and we can see we can help you

**Comment by dineshkumar-kanniyammal** [comment]
> 
**how are you running in parallel, how many sessions etc**
1. 600 sessions using 
2. used Selenium 4.1.2 and Selenium 4.0.0

I am facing an issue of  org.openqa.selenium.NoSuchSessionException:Tried to run command without establishing a connection

@fhoeben
